### ISSUE TITLE
Unclear Licensing Terms

### ISSUE THREAD

**Issue by MCLx86** [BetterThanUpdates/apron#38]
Hello,
the Forge License has the following line:
"As a prerequisite, a **user of Minecraft Forge must own a legally acquired copy of Minecraft**",
which infringes upon Freedom 0.

I recommend rewriting all the Forge code and releasing it under a license that is actually free.

**Comment by halotroop2288** [comment]
There are no contradictions in our licensing. Minecraft Forge and any modifications we have made to Minecraft Forge are released under a separate license to our original code (in the `io.github.betterthanupdates` package).
If this is at all unclear, please let us know how we can make it more clear.

As it stands, we have bigger licensing issues we're trying to solve. Like getting official permission to use other libraries' code.
At this time, we will comply with any official take-down request submitted by original authors and copyright holders, for code that does not have its own notice of license attached.

**Comment by MCLx86** [comment]
I understand it's a separate license, but that doesn't change the fact it's non-free (i.e. it doesn't fully guarantee the 4 Freedoms). Now, I do realize that this is not that big of a deal, but nevertheless it makes the software non-free and incompatible with Free (open source/libre) licenses.

**Comment by halotroop2288** [comment]
By default, all Minecraft mods require the user to own a "legally" licensed copy of the game, as per the Minecraft EULA. Forge retains this notice because it was originally written as patches to the main game and could not be abstracted easily. And we did not touch the license except to fix obvious spelling mistakes. It still maintains the exact spirit of the license as the original authors agreed to.
Fabric fixes this by letting us use Mixin. So *technically* you could abstract it all away and link this mod to another game entirely.

I imagine this is a similar conversation that they had when they changed their own license to GPL (which is *also* incompatible with the Minecraft EULA, and still remains that way to this day).
Regardless, we're not re-licensing or rewriting Minecraft Forge at this time. If it's semantically not considered "free" because of that, that's just fine with me.

In the future, we might actually abstract *all* the code away from Forge classes to our own API. At least, that was in *my* plans. I don't know what @thecatcore will do with it as the maintainer.

---

TL;DR: Semantics. Don't care if it's not considered "free" technically.

**Comment by MCLx86** [comment]
> By default, all Minecraft mods require the user to own a "legally" licensed copy of the game, as per the Minecraft EULA. Forge retains this notice because it was originally written as patches to the main game and could not be abstracted easily. And we did not touch the license except to fix obvious spelling mistakes. It still maintains the exact spirit of the license as the original authors agreed to. Fabric fixes this by letting us use Mixin. So _technically_ you could abstract it all away and link this mod to another game entirely.
> 
> I imagine this is a similar conversation that they had when they changed their own license to GPL (which is _also_ incompatible with the Minecraft EULA, and still remains that way to this day). Regardless, we're not re-licensing or rewriting Minecraft Forge at this time. If it's semantically not considered "free" because of that, that's just fine with me.
> 
> In the future, we might actually abstract _all_ the code away from Forge classes to our own API. At least, that was in _my_ plans. I don't know what @arthurbambou will do with it as the maintainer.
> 
> TL;DR: Semantics. Don't care if it's not considered "free" technically.

I am not a lawyer, and I might be wrong about this, but from what I know, mods don't require the user to own a licensed copy of the game, as they are considered original works and do not contain any of Minecraft's code, therefore the EULA does not apply to them.
